### PR TITLE
Polish of writing smart contracts with rust

### LINKED ISSUE #17: Polish "Writing Smart Contract with Rust"

### PR DESCRIPTION
**Pull request by KMCreatesWorlds**
Added blocks:
- Tip for using the automatic project structure
- Info for important syntax from Rust to be read before proceeding further

### CODE REVIEW

File: source/docs/casper/dapp-dev-guide/writing-contracts/rust-contracts.md [hunk 1]
```diff
@@ -22,6 +22,15 @@ The contract contains required metadata, and it is primarily identified by its `
...
 
+:::tip
+
+As a beginner it is not advised to start with the manual project structure. 
+Structure created automatically with `cargo casper` contains everything that is needed to start coding.
+
+:::
+
+The project structure will be different while designing the full stack architecture. This will be expanded upon while describing the dApps.
+
```
Comment (ghost) [inline source/docs/casper/dapp-dev-guide/writing-contracts/rust-contracts.md]: Tip with additional sentence breaks the "below" reference:

image

I would reorganize it a little bit, maybe put structure before tip :thinking:.

File: source/docs/casper/dapp-dev-guide/writing-contracts/rust-contracts.md [hunk 2]
```diff
@@ -84,6 +93,21 @@ The `cargo casper` command can automatically set up the project structure, as sh
...
+:::info
+
+Important syntax elements used frequently in Rust:
+
+[Match](https://doc.rust-lang.org/rust-by-example/flow_control/match.html)<br />
+[Array](https://doc.rust-lang.org/rust-by-example/primitives/array.html)<br />
+[Loop](https://doc.rust-lang.org/rust-by-example/flow_control/loop.html)<br />
+[Vectors](https://doc.rust-lang.org/rust-by-example/std/vec.html)<br />
+[Functions](https://doc.rust-lang.org/rust-by-example/fn.html)<br />
+[Generics](https://doc.rust-lang.org/rust-by-example/generics.html)<br />
```
Comment (ghost) [inline source/docs/casper/dapp-dev-guide/writing-contracts/rust-contracts.md]: I think generics are not common in smart contracts and it is rather difficult topic for beginners. I would advise to remove that line.

File: source/docs/casper/dapp-dev-guide/writing-contracts/rust-contracts.md [hunk 3]
```diff
@@ -84,6 +93,21 @@ The `cargo casper` command can automatically set up the project structure, as sh
 
 This section covers the process of writing a smart contract in Rust, using example code from the [counter contract](https://github.com/casper-ecosystem/counter/). This simple contract allows callers to increment and retrieve an integer. Casper provides a [contract API](https://docs.rs/casper-contract/latest/casper_contract/contract_api/index.html) within the [`casper_contract`](https://docs.rs/casper-contract/latest/casper_contract/index.html) crate.
 
+:::info
+
+Important syntax elements used frequently in Rust:
+
+[Match](https://doc.rust-lang.org/rust-by-example/flow_control/match.html)<br />
```
Comment (ghost) [inline source/docs/casper/dapp-dev-guide/writing-contracts/rust-contracts.md]: I would use bullet points to list syntax elements.

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]
I think the original file needs some rewriting, however these changes are good so I am approving.

File: source/docs/casper/dapp-dev-guide/writing-contracts/rust-contracts.md (hunk 1 above)
Comment (ghost) [reply source/docs/casper/dapp-dev-guide/writing-contracts/rust-contracts.md]: Yes, in general tips are better placed *after* the piece of information they refer to. It is disorienting to read a "tip" when you don't yet have awareness of the concept the tip is referring to.

**Comment by KMCreatesWorlds** [comment]
Made the following changes according to the review:
- The "Tip" moved after the Manual and Automatic creation
- Generics removed from the list
- Bullet points added to the list

File: source/docs/casper/dapp-dev-guide/writing-contracts/rust-contracts.md (hunk 1 above)
Comment (KMCreatesWorlds) [reply source/docs/casper/dapp-dev-guide/writing-contracts/rust-contracts.md]: Changes made in the new commit.

File: source/docs/casper/dapp-dev-guide/writing-contracts/rust-contracts.md (hunk 2 above)
Comment (KMCreatesWorlds) [reply source/docs/casper/dapp-dev-guide/writing-contracts/rust-contracts.md]: Changes made in the new commit.

File: source/docs/casper/dapp-dev-guide/writing-contracts/rust-contracts.md (hunk 3 above)
Comment (KMCreatesWorlds) [reply source/docs/casper/dapp-dev-guide/writing-contracts/rust-contracts.md]: Changes made in a new commit.

**Review by ghost — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**